CRM/Pledge/BAO/Query.php

```diff
@@ -82,6 +82,12 @@ public static function select(&$query) {
       $query->_tables['civicrm_pledge'] = $query->_whereTables['civicrm_pledge'] = 1;
     }
 
+    if (!empty($query->_returnProperties['pledge_end_date'])) {
```

Comment:
hmm this shouldn't be needed - I will look & try to see why it was

Reply:
This was how i fixed an issue with the api_v3_SyntaxConformanceTest

Reply:
OK - it's in SELECT not WHERE - which I guess is  where we've been focussing cleanup